datafusion/optimizer/src/simplify_expressions.rs

```diff
@@ -1573,13 +1597,11 @@ mod tests {
             low: Box::new(lit(0)),
             high: Box::new(lit(10)),
         };
+        let between_expr = expr.clone();
         let expr = expr.or(lit_bool_null());
         let result = simplify(expr);
 
-        let expected_expr = or(
-            and(col("c1").gt_eq(lit(0)), col("c1").lt_eq(lit(10))),
-            lit_bool_null(),
-        );
+        let expected_expr = or(between_expr, lit_bool_null());
```

Comment:
```suggestion
        let expected_expr = or(between_expr, lit_bool_null());
        // let expected_expr = or(
        //    and(col("c1").gt_eq(lit(0)), col("c1").lt_eq(lit(10))),
        //    lit_bool_null(),
        //);        
```

Reply:
I suggest keeping the old expected result in the code as comments to make it easier to understand what the desired effect is